manager/controllers/app/m4dapplication_controller.go

```diff
@@ -69,8 +72,9 @@ func (r *M4DApplicationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 	observedStatus := applicationContext.Status.DeepCopy()
 
 	// check if reconcile is required
-	generatedBlueprint, _ := r.getBlueprint(applicationContext)
-	if generatedBlueprint == nil || observedStatus.ObservedGeneration != applicationContext.GetGeneration() {
+	// reconcile is required if the spec has been changed, or no blueprint generation was done (reconcile due to non-fatal errors)
```

Comment:
Comment missing plotter.  While generalization to resource which can be either a blueprint or a plotter is good, it makes the code harder to understand so the comments are important

Reply:
Updated comments.